File: IAPCR2F.cpp
#include<bits/stdc++.h>
using namespace std;
vector<int> adj[1001];
vector<int> cost;
int visit[1001];
void dfs(int s,int &sum){
  visit[s]=true;
  sum+=cost[s-1];
  for(int i=0;i<adj[s].size();i++)
  {
    if(!visit[adj[s][i]])
    dfs(adj[s][i],sum);
  }
}
int main()
{
  int t;
  cin>>t;
  for(int T=1;T<=t;T++)
  {
    for(int i=0;i<1001;i++)
    adj[i].clear();
    cost.clear();
    int n,m;
    cin>>n>>m;
    fill(visit,visit+n+1,false);
    for(int i=0;i<n;i++){
      int v;
      cin>>v;
      cost.push_back(v);
    }
    while(m--){
      int u,v;
      cin>>u>>v;
      adj[u].push_back(v);
      adj[v].push_back(u);
    }
    vector<int> res;
    int sum=0,count=0;

    for(int i=1;i<=n;i++)
    {
      sum=0;
      if(!visit[i]){
      dfs(i,sum);
      res.push_back(sum);
      count++;
      }
    }
    sort(res.begin(),res.end());
    cout<<"Case "<<T<<": "<<count<<endl;
    for(int i=0;i<res.size();i++)
    cout<<res[i]<<" ";
    cout<<endl;


  }
}

File: Codeforces/src/IAPCR2F.py
a=[[]]
def dfs(v,visit,val):
	print(a[v])
	visit[v]=True
	val+=cost[v-1]
	for i in a[v]:
		#print(i)
		if visit[i]==False:
			dfs(i,visit,val)
if __name__ == '__main__':
	t = int(input())
	for T in range(1,t+1):
		visit = [False]*1001
		n,m = map(int,input().split())
		for i in range(1,n+1):
			a.append(i)
		cost = list(map(int,input().split()))
		for i in range(m):
			u,v = map(int,input().split())
			a[u].append(v)
			a[v].append(u)
		print(a[2])
		sum=[]
		count=0
		visit = [False]*1001
		for i in range(1,n+1):
			if visit[i]==False:
				#print(i)
				val=0
				dfs(i,visit,val)
				sum.append(val)
				count+=1
		print("Case "+str(T)+": "+str(count))
		print(" ".join(str(i) for i in sum))
		a.clear()
		

		


  